flink-test-utils-parent/flink-test-utils/src/main/java/org/apache/flink/streaming/util/PseudoRandomValueSelector.java

```diff
@@ -0,0 +1,122 @@
+ * cases. Thus, two different builds will mostly result in different values for the same test case.
+ * Similarly, two test cases in the same build will have different randomized values.
+ *
+ * <p>The seed can be set with the maven/system property test.randomization.seed and is set by
+ * default to commit id. If the seed is empty, {@link EnvironmentInformation} and as a last fallback
+ * git command is used to retrieve the commit id.
+ */
+@Internal
+@NotThreadSafe
+class PseudoRandomValueSelector {
+    private final Function<Integer, Integer> randomValueSupplier;
+
+    private static final Logger LOG = LoggerFactory.getLogger(PseudoRandomValueSelector.class);
+
+    private static final long GlobalSeed = (long) getGlobalSeed().hashCode() << 32;
+
+    private PseudoRandomValueSelector(Function<Integer, Integer> randomValueSupplier) {
+        this.randomValueSupplier = randomValueSupplier;
+    }
+
+    public <T> void select(Configuration configuration, ConfigOption<T> option, T... alternatives) {
+        if (configuration.contains(option)) {
+            return;
+        }
+        final Integer choice = randomValueSupplier.apply(alternatives.length);
+        T value = alternatives[choice];
+        LOG.info("Randomly selected {} for {}", value, option.key());
+        configuration.set(option, value);
+    }
+
+    public static PseudoRandomValueSelector create(Object entryPointSeed) {
+        final long combinedSeed = GlobalSeed | entryPointSeed.hashCode();
+        final Random random = new Random(combinedSeed);
+        return new PseudoRandomValueSelector(random::nextInt);
+    }
+
+    private static String getGlobalSeed() {
+        // manual seed or set by maven
+        final String seed = System.getProperty("test.randomization.seed");
+        if (seed != null) {
+            return seed;
+        }
+
+        // Read with git command (if installed)
+        final Optional<String> gitCommitId = getGitCommitId();
+        if (gitCommitId.isPresent()) {
+            return gitCommitId.get();
+        }
+
+        // try EnvironmentInformation, which is set in the maven process
+        final String commitId = EnvironmentInformation.getGitCommitId();
+        if (!commitId.equals(EnvironmentInformation.UNKNOWN_COMMIT_ID)) {
+            return commitId;
+        }
+
+        LOG.warn(
+                "Cannot initialize maven property test.randomization.seed with commit id, please set manually to receive reproducible builds.");
+        // return any constant
+        return "";
+    }
+
+    @VisibleForTesting
+    static Optional<String> getGitCommitId() {
+        try {
+            Process process = new ProcessBuilder("git", "rev-parse", "HEAD").start();
+            try (InputStream input = process.getInputStream()) {
+                final String commit = IOUtils.toString(input, Charset.defaultCharset()).trim();
+                if (commit.matches("[a-f0-9]{40}")) return Optional.of(commit);
```

Comment:
braces! (I hope this will fail the CI, if not we might have to re-enable some checkstyle rules...)